apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -808,21 +977,50 @@ impl SpanExporter for Exporter {
                 self.spans_by_parent_id
                     .get_mut(&span.parent_span_id)
                     .expect("capacity of cache was zero")
-                    .push(len, span.into());
+                    .push(len, LightSpanData::from_span_data(span, &self.include_attr_names));
             }
         }
         tracing::info!(value.apollo_router_span_lru_size = self.spans_by_parent_id.len() as u64,);
         let mut report = telemetry::apollo::Report::default();
         report += SingleReport::Traces(TracesReport { traces });
-        let exporter = self.report_exporter.clone();
+        #[allow(clippy::manual_map)] // https://github.com/rust-lang/rust-clippy/issues/8346
+        let report_exporter = match self.report_exporter.as_ref() {
+            Some(exporter) => Some(exporter.clone()),
+            None => None,
+        };
+        #[allow(clippy::manual_map)] // https://github.com/rust-lang/rust-clippy/issues/8346
+        let otlp_exporter = match self.otlp_exporter.as_ref() {
+            Some(exporter) => Some(exporter.clone()),
+            None => None,
+        };
+
         let fut = async move {
-            exporter
-                .submit_report(report)
-                .map_err(|e| TraceError::ExportFailed(Box::new(e)))
-                .await
+            let mut exports: Vec<BoxFuture<ExportResult>> = Vec::new();
+            if let Some(exporter) = report_exporter.as_ref() {
+                exports.push(
+                    exporter
+                        .submit_report(report)
+                        .map_err(|e| TraceError::ExportFailed(Box::new(e)))
+                        .boxed(),
+                );
+            }
+            if let Some(exporter) = otlp_exporter.as_ref() {
+                exports.push(exporter.export(otlp_trace_spans.into_iter().flatten().collect()));
+            }
+            match try_join_all(exports).await {
+                Ok(_) => Ok(()),
+                Err(e) => Err(e),
+            }
         };
         fut.boxed()
     }
+
+    fn shutdown(&mut self) {
+        // currently only handled in the OTLP case
+        if let Some(exporter) = self.otlp_exporter.clone() {
```

Comment:
```suggestion
        if let Some(exporter) = &self.otlp_exporter {
```
or you can `.take()` it if you don't need it anymore ?

Reply:
So far no issues with this suggestion.